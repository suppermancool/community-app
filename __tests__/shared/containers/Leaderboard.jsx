import _ from 'lodash';
import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';

const rnd = new Rnd();

const mockUserObj = {
  rank: 1,
  avatarUrl: 'some.fake.string',
  handle: 'overachieving.mofo',
  challengecount: 99999,
  'project_result.final_score': 9999999999,
};
const mockLeaderboardData = [mockUserObj, mockUserObj, mockUserObj, mockUserObj];

const mockLeaderboardActions = {
  leaderboard: {
    fetchLeaderboardInit: jest.fn(),
    fetchLeaderboardDone: jest.fn(() => mockLeaderboardData),
  },
};
jest.setMock(require.resolve('actions/leaderboard'), mockLeaderboardActions);

const mockState = {
  leaderboard: {
    data: mockLeaderboardData,
    loading: false,
  },
};

const mockState2 = {
  leaderboard: {},
};

const Leaderboard = require('containers/Leaderboard').default;

beforeEach(() => jest.clearAllMocks());

test('Matches shapshot', () => {
  rnd.render((
    <Leaderboard
      store={{
        dispatch: () => _.noop,
        getState: () => mockState,
        subscribe: _.noop,
      }}
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});

test('Triggers data loading, if necessary', () => {
  TU.renderIntoDocument((
    <Leaderboard
      store={{
        dispatch: () => _.noop,
        getState: () => mockState2,
        subscribe: _.noop,
      }}
    />
  ));
  expect(mockLeaderboardActions.leaderboard.fetchLeaderboardInit).toHaveBeenCalled();
  expect(mockLeaderboardActions.leaderboard.fetchLeaderboardDone).toHaveBeenCalled();
});

const obj = TU.renderIntoDocument((
  <Leaderboard
    store={{
      dispatch: () => _.noop,
      getState: () => mockState,
      subscribe: _.noop,
    }}
  />
));

const { props } = obj.selector;

test('loadLeaderboard dispatch', () => {
  const { loadLeaderboard } = props;
  loadLeaderboard();
  expect(mockLeaderboardActions.leaderboard.fetchLeaderboardInit).toHaveBeenCalled();
  expect(mockLeaderboardActions.leaderboard.fetchLeaderboardDone).toHaveBeenCalled();
});
