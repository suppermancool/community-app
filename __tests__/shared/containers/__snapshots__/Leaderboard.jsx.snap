// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shapshot 1`] = `
<LeaderboardPageContainer
  HeadBanner={null}
  apiUrl="http://www.mocky.io/v2/59098e60100000b60747c10b"
  auth={null}
  isLoadingLeaderboard={false}
  leaderboardData={
    Array [
      Object {
        "avatarUrl": "some.fake.string",
        "challengecount": 99999,
        "handle": "overachieving.mofo",
        "project_result.final_score": 9999999999,
        "rank": 1,
      },
      Object {
        "avatarUrl": "some.fake.string",
        "challengecount": 99999,
        "handle": "overachieving.mofo",
        "project_result.final_score": 9999999999,
        "rank": 1,
      },
      Object {
        "avatarUrl": "some.fake.string",
        "challengecount": 99999,
        "handle": "overachieving.mofo",
        "project_result.final_score": 9999999999,
        "rank": 1,
      },
      Object {
        "avatarUrl": "some.fake.string",
        "challengecount": 99999,
        "handle": "overachieving.mofo",
        "project_result.final_score": 9999999999,
        "rank": 1,
      },
    ]
  }
  loadLeaderboard={[Function]}
  loadedApiUrl={null}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
